src/derivative_wrappers.jl

```diff
@@ -37,19 +37,13 @@ function derivative(f, x::Union{Number,AbstractArray{<:Number}},
 end
 
 jacobian_autodiff(f, x, odefun) = (ForwardDiff.derivative(f,x),1)
-function jacobian_autodiff(f, x::AbstractArray, odefun)
-  if DiffEqBase.has_colorvec(odefun)
-    colorvec = odefun.colorvec
-    sparsity = odefun.jac_prototype
-    jac_prototype = nothing
-  else
-    colorvec = 1:length(x)
-    sparsity = nothing
-    jac_prototype = odefun.jac_prototype
-  end
+function jacobian_autodiff(f, x::AbstractArray, odefun, chunk_size)
+  colorvec = DiffEqBase.has_colorvec(odefun) ? odefun.colorvec : 1:length(x)
+  sparsity = odefun.sparsity
+  jac_prototype = odefun.jac_prototype
   maxcolor = maximum(colorvec)
-  chunksize = getsize(default_chunk_size(maxcolor))
-  num_of_chunks = Int(ceil(maxcolor / chunksize))
+  chunk_size = chunk_size==0 ? getsize(default_chunk_size(maxcolor)) : chunk_size
```

Comment:
this accepts Val?

Reply:
Yeah, need to wrap it with getsize

Reply:
I don't follow here. Is the code as it stands now ok?